lib/gssapi/krb5/import_name.c

```diff
@@ -149,15 +149,12 @@ import_hostbased_name(OM_uint32 *minor_status,
 	host = p + 1;
     }
 
-    kerr = krb5_make_principal(context, &princ, "", tmp, host, NULL);
+    kerr = krb5_sname_to_principal(context, host, tmp, KRB5_NT_SRV_HST, &princ);
```

Comment:
Well, I'm going to note you need to rethink this :)

Reply:
Oh?  Why?

Reply:
To avoid immediate canonicalization?  With the name rules stuff you get that for free, though it's not the default (but maybe it ought to be).